test/agent.js

```diff
@@ -789,31 +792,36 @@ test('#flush()', function (t) {
         this.agent.startTransaction('foo')
         this.agent.endTransaction()
         this.agent.flush(function (err) {
-          t.error(err)
+          // 2. ... the flush callback should be called.
+          t.error(err, 'no error passed to agent.flush callback')
           t.pass('should call flush callback after flushing the queue')
+          t.end()
         })
       })
       .on('data-transaction', function (data) {
-        t.strictEqual(data.name, 'foo')
-        t.end()
+        // 1. The APM server should receive the transaction, and then ...
+        t.strictEqual(data.name, 'foo', 'APM server received the "foo" transaction')
       })
```

Comment:
Reviewer note: All of the test fixes are of this form. Now that the http client actually waits for a response from the APM server before concluding, the callback to `.flush(cb)` or to `.captureError(cb)` is reliably called *after* the APM server receives the data (i.e. the "data-transaction" event from the mock APM server here).